pbs/usersync.go

```diff
@@ -10,33 +10,86 @@ import (
 	"strings"
 	"time"
 
+	"errors"
 	"github.com/golang/glog"
 	"github.com/julienschmidt/httprouter"
 	"github.com/prebid/prebid-server/ssl"
-	metrics "github.com/rcrowley/go-metrics"
+	"github.com/rcrowley/go-metrics"
 )
 
-var cookie_domain string
-var external_url string
-var recaptcha_secret string
+// Recaptcha code from https://github.com/haisum/recaptcha/blob/master/recaptcha.go
+const RECAPTCHA_URL = "https://www.google.com/recaptcha/api/siteverify"
+const COOKIE_NAME = "uids"
 
-type PBSCookie struct {
-	UIDs     map[string]string `json:"uids,omitempty"`
-	OptOut   bool              `json:"optout,omitempty"`
-	Birthday *time.Time        `json:"bday,omitempty"`
+const (
+	USERSYNC_OPT_OUT     = "usersync.opt_outs"
+	USERSYNC_BAD_REQUEST = "usersync.bad_requests"
+	USERSYNC_SUCCESS     = "usersync.%s.sets"
+)
+
+type UserSyncDeps struct {
+	Cookie_domain    string
+	External_url     string
+	Recaptcha_secret string
+	Metrics          metrics.Registry
+}
+
+// UserSyncMap is cookie which stores the user sync info for all of our bidders.
+//
+// To get an instance of this from a request, use ParseUserSyncMapFromRequest.
+// To write an instance onto a response, use SetCookieOnResponse.
+type UserSyncMap interface {
```

Comment:
Is it possible instead of having the interface to create a struct that is public just for the purpose of JSON serialization? Something like what is described here (https://stackoverflow.com/questions/28035784/golang-marshal-unmarshal-json-with-both-exported-and-un-exported-fields).

Would that allow the `uuids` field to be private still without the need for the interface?